src/hooks.ts

```diff
@@ -36,3 +42,113 @@ export const useHasResourcePermission = (resource: Resource, authzUrl: string, p
     const { permissions, isFetching } = useResourcePermissions(resource, authzUrl) ?? {};
     return { isFetching, hasPermission: permissions.includes(permission) };
 };
+
+export const useOpenIdConfig = (openIdConfigUrl: string) => {
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        dispatch(fetchOpenIdConfiguration(openIdConfigUrl));
+    }, [dispatch, openIdConfigUrl]);
+
+    return useSelector((state: RootState) => state.openIdConfiguration.data);
+}
+
+export const useSignInPopupTokenHandoff = (
+    applicationUrl: string,
+    authCallbackUrl: string,
+    clientId: string,
+    windowMessageHandler: MutableRefObject<null | MessageHandlerFunc>
+) => {
+    const dispatch = useDispatch();
+    const removeListenerIfNeeded = () => {
+        if (windowMessageHandler.current) {
+            window.removeEventListener("message", windowMessageHandler.current);
+        }
+    }
+    useEffect(() => {
+        removeListenerIfNeeded();
```

Comment:
this shouldn't be needed, since cleanup will run before the effect is called again